lib/modules/apostrophe-palette-global/index.js

```diff
@@ -129,13 +127,53 @@ module.exports = {
       {
         name: 'palette',
         label: 'Palette Fields',
-        fields: fieldNames
+        fields: fieldNames.concat(['paletteCounter'])
       }
     ]);
 
+    // This route serves the existing palette stylesheet, constructed from the global object
+    // We do it this way so the browser can cache the styles as often as possible
+    self.route('get', 'palette-styles', function (req, res) {
+      var paletteFields = self.apos.modules['apostrophe-global'].options.paletteFields;
+      var rules = [];
+
+      paletteFields.forEach(function(field) {
+        var selectors = field.selector;
+        var properties = field.property;
+        var fieldValue = req.data.global[field.name];
+        var fieldUnit = field.unit || '';
+
+        if (_.isString(selectors)) {
+          selectors = [ selectors ];
+        }
+
+        if (_.isString(properties)) {
+          properties = [ properties ];
+        }
+
+        properties.forEach(function (property) {
+          selectors.forEach(function (selector) {
+            var rule = '';
+            if (field.mediaQuery) {
+              rule += '@media ' + field.mediaQuery + ' { ';
+            }
+            rule += selector + ' { ' + property + ': ' + fieldValue + fieldUnit + '; }';
+            if (field.mediaQuery) {
+              rule += ' } ';
+            }
+            rules.push(rule);
+          });
+        });
+      });
+
+      res.setHeader('Content-Type', 'text/css');
```

Comment:
We are using a GET route with a URL that changes with each release and relying on the browser to cache it. That's a good direction, but I recommend you send caching headers to make sure it caches it for a long time.

One year would be:

```
res.setHeader('Cache-Control', 'public, max-age=31557600');
```